packages/sit-onyx/src/components/OnyxDataGrid/OnyxDataGrid.vue

```diff
@@ -8,28 +12,43 @@ import {
   type DataGridRendererRow,
   type OnyxDataGridProps,
 } from "../..";
-import { useDataGridFeatures } from "./features";
-import { useDataGridSorting } from "./features/sorting/sorting";
+import { useDataGridFeatures, type DataGridFeature } from "./features";
 
-const props = defineProps<OnyxDataGridProps<TEntry>>();
-
-const withSorting = useDataGridSorting<TEntry>();
-
-const { watchSources, createRendererRows, createRendererColumns } = useDataGridFeatures([
-  withSorting,
-]);
+const props = withDefaults(defineProps<OnyxDataGridProps<TEntry, TFeatures>>(), {
+  features: () => [] as TFeatures,
+});
 
 // Using Ref types to avoid `UnwrapRef` issues
 const renderColumns: Ref<DataGridRendererColumn<TEntry, object>[]> = ref([]);
 const renderRows: Ref<DataGridRendererRow<TEntry, DataGridMetadata>[]> = ref([]);
 
-const { columns, data } = toRefs(props);
+const { columns, data, features } = toRefs(props);
+
+const featureBuilder = computed<ReturnType<typeof useDataGridFeatures<TEntry, TFeatures>>>(() =>
```

Comment:
```suggestion
const featureBuilder = computed<ReturnType<typeof useDataGridFeatures<TEntry, TFeatures>>>(() =>
```

Is this type needed? Shouldn't it be inferred automatically?

Reply:
it doesn't infer the generic parts...